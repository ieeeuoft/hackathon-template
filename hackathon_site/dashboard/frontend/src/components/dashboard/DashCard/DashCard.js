import React from "react";
import styles from "./DashCard.module.scss";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import TitledPaper from "components/general/TitledPaper/TitledPaper";

const CardItem = ({ name, icon }) => (
    <Container className={styles.items}>
        <Typography variant="body2" noWrap>
            {name}
        </Typography>
        {icon}
    </Container>
);

const DashCard = ({ title, content }) => (
    <TitledPaper title={title}>
        {content.map((listItem, i) => (
            <Link
                href={listItem.url}
                rel="noopener"
                color="inherit"
                underline="none"
                target="_blank"
                key={i}
            >
                <CardItem name={listItem.name} icon={listItem.icon} />
            </Link>
        ))}
    </TitledPaper>
);

export default DashCard;
