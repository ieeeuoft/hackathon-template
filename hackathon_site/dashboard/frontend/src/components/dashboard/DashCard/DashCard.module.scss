@import "./../../../assets/abstracts/mixins";
@import "./../../../assets/abstracts/variables";

.items {
    @include flexPosition($jusCon: space-between, $alIte: center);
    padding-top: 10px;
    padding-bottom: 10px;
    &:hover {
        background-color: color(light2);
    }
    svg {
        margin-left: 15px;
    }
}

.paper::-webkit-scrollbar {
    display: none;
}
