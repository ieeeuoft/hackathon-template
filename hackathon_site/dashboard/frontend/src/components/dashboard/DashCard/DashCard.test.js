import React from "react";
import DashCard from "./DashCard";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("DashCard", () => {
    const testTitle = "any string";
    const testContent = [
        {
            name: "testing",
            url: "https://jestjs.io/docs/en/getting-started",
            icon: null,
        },
        { name: "testing", url: "youtube", icon: null },
    ];

    test("render", () => {
        const { asFragment } = render(
            <DashCard title={testTitle} content={testContent} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
