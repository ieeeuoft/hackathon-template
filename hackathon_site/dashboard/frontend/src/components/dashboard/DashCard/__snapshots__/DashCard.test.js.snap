// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashCard render 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root title MuiTypography-h2 MuiTypography-noWrap"
    data-testid="titledPaperTitle"
  >
    any string
  </h2>
  <div
    class="MuiPaper-root paper MuiPaper-elevation2"
  >
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
      href="https://jestjs.io/docs/en/getting-started"
      rel="noopener"
      target="_blank"
    >
      <div
        class="MuiContainer-root items MuiContainer-maxWidthLg"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
        >
          testing
        </p>
      </div>
    </a>
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
      href="youtube"
      rel="noopener"
      target="_blank"
    >
      <div
        class="MuiContainer-root items MuiContainer-maxWidthLg"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
        >
          testing
        </p>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
